app/services/proofing/resolution/resolution_context.rb

```diff
@@ -0,0 +1,19 @@
+# frozen_string_literal: true
+
+module Proofing
+  module Resolution
+    class ResolutionContext
+      ## A container for resolution, pii, user_email, config store
+      attr_reader :pii, :app_config_store
+      attr_accessor :user_email
+
+      # @param [Pii::Attributes] pii
+      # @param [String] user_email
+      def initialize(pii:, user_email: nil)
+        @pii = pii
+        @user_email = user_email
+        @app_config_store = IdentityConfig.store
```

Comment:
What is the purpose of saving this here? It's not injected into the constructor, so we'll still need to stub it out in specs?

Same goes for this context object, I'm not convinced we need it?

Reply:
yeah, originally planned to use it.

Right now the IdentityConfig load items from files, let's say later we may wanna to load all config items from some key value store, we potentially can inject a class of the config store without changing client code. Also you can mock the store anyway you want.

The resolutioncontext is to think at the higher level, along with proofers. Proofers's only responsibility is to proof, either it's a mock, or client to different vendors.

What information a proofer need to do it's job? I would say pii(+user_id/email) and the configuration of the application. So resolution context provide a common container to provide needed information for proofer to do its job.

Reply:
IdentityConfig is already our intermediate layer.  In the past, we've pulled from ENV vars directly scattered throughout the codebase. Today, yes, IdentityConfig only pulls from files, but the interface is already standard enough that it could pull other places if/when we have those. I would not add this layer of indirection yet